test/unit/ngsiv2/provisioning/device-registration_test.js

```diff
@@ -0,0 +1,344 @@
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/registrations', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextAvailabilityRequests/registerIoTAgent1.json'))
+                .reply(404);
+
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                iotAgentLib.clearAll(done);
+            });
+        });
+
+        it('should register as ContextProvider of its lazy attributes', function(done) {
+            iotAgentLib.register(device1, function(error) {
+                should.exist(error);
+                error.name.should.equal('BAD_REQUEST');
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When the Context Broker returns an HTTP transport error while registering a device', function() {
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/registrations', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextAvailabilityRequests/registerIoTAgent1.json'))
+                .reply(500);
+
+            iotAgentLib.activate(iotAgentConfig, function(error) {
+                iotAgentLib.clearAll(done);
+            });
+        });
+
+        it('should not register the device in the internal registry');
+        it('should return a REGISTRATION_ERROR error to the caller', function(done) {
+            iotAgentLib.register(device1, function(error) {
+                should.exist(error);
+                should.exist(error.name);
+                error.name.should.equal('REGISTRATION_ERROR');
+
+                done();
+            });
+        });
+    });
+
+    describe('When a device is requested to the library using its ID', function() {
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/registrations', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextAvailabilityRequests/registerIoTAgent1.json'))
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5